app/src/main/java/com/github/h3lp3rs/h3lp/MainPageActivity.kt

```diff
@@ -72,6 +102,100 @@ class MainPageActivity : AppCompatActivity(), OnRequestPermissionsResultCallback
         if (checkSelfPermission(this, ACCESS_FINE_LOCATION) != PERMISSION_GRANTED) {
             requestPermissions(arrayOf(ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST_CODE)
         }
+
+        startAppGuide()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        // This removes the focus from the search bar when the user goes back to the main page
+        // after clicking on an item from the list view.
+        searchView.clearFocus()
+    }
+
+    /**
+     * Starts the application guide. Uses preferences to only start the guide when the app is
+     * launched for the first time.
+     */
+    private fun startAppGuide() {
+        val prefManager = PreferenceManager.getDefaultSharedPreferences(this)
+        if (!prefManager.getBoolean("didShowGuide", false)) {
+            val prefEditor = prefManager.edit()
```

Comment:
Depending on the merge order, this will have to be modified in accordance with the new local storage